src/node/wrapper.ts

```diff
@@ -111,19 +164,15 @@ class ChildProcess extends Process {
   /**
    * Initiate the handshake and wait for a response from the parent.
    */
-  public handshake(): Promise<void> {
-    return new Promise((resolve) => {
-      const onMessage = (message: Message): void => {
-        logger.debug(`received message from ${this.parentPid}`, field("message", message))
-        if (message.type === "handshake") {
-          process.removeListener("message", onMessage)
-          resolve()
-        }
-      }
-      // Initiate the handshake and wait for the reply.
-      process.on("message", onMessage)
-      this.send({ type: "handshake" })
-    })
+  public async handshake(): Promise<void> {
+    this.send({ type: "handshake" })
```

Comment:
Shouldn't we register the handler before sending? To avoid the message being lost.

Reply:
I don't think it's possible since it's single-threaded so nothing will be able to run in between the send and when we attach the message listener. I'm down to change it just in case though!

Reply:
Fair! let's leave as is then.